lib/main_common.pm

```diff
@@ -95,7 +95,7 @@ sub set_defaults_for_username_and_password {
         $testapi::password = '';
     }
     else {
-        if (get_var('FLAVOR', '') =~ /SAP/) {
+        if (get_var('FLAVOR', '') =~ /SAP/ or get_var('SLE_PRODUCT', '') =~ /sles4sap/) {
```

Comment:
Is SLE_PRODUCT doesn't exist in SLE12 as well? To avoid the use of the `get_car('FLAVOR',...)`.
Other than this, LGTM :-)

Reply:
As I understood it, `SLE_PRODUCT` was introduced for `SLE-15` testing to handle the single media, but I don't see any reason why it couldn't be used with prior versions as well. In fact, as I received the tests, the way to install `SLES` using the sles4sap media is with `FLAVOR=SAP-DVD` and `SLES4SAP_MODE=sles` and that's how it's currently coded. I guess we could change that so the same can be achieved with `SLE_PRODUCT=sles` and `FLAVOR=SAP-DVD`, but I don't see the need for it. Keeping the `or`, give us backward compatibility with the tests.

Reply:
Ok for me :-)